agent/src/test/java/com/teamscale/jacoco/agent/upload/delay/DelayedCommitDescriptorRetrievalTest.java

```diff
@@ -18,11 +20,11 @@
 	@Test
 	public void locatorShouldTriggerUploadOfCachedXmls() throws Exception {
 		ExecutorService storeExecutor = Executors.newSingleThreadExecutor();
-		File coverageFile = File.createTempFile("jacoco-", ".xml");
-		Path outputPath = coverageFile.getParentFile().toPath();
+		CoverageFile coverageFile = new CoverageFile(File.createTempFile("jacoco-", ".xml"));
```

Comment:
I'd prefer you create a temp *dir*, then create a file with a fixed name inside, as you're later iterating over the temp dir which may contain leftovers from other tests

Reply:
same for the other tests where you use `getParentDirectoryPath`